buckets/redis/bucket.go

```diff
@@ -67,6 +67,19 @@ func (bf *bucketFactory) Init(cfg *pbconfig.ServiceConfig) {
 	if bf.client == nil {
 		bf.connectToRedisLocked()
 	}
+
+	if err := bf.cleanStaleBuckets() ; err != nil {
```

Comment:
I benchmarked this, clearing tens of millions of buckets (spread across thousands of namespaces) took single-digit seconds. So that would affect startup and how quickly new configs take effect accordingly, but for the sake of simplifying update overlaps as you mentioned, I don't think it's that bad.

Alternatively, `FLUSHDB` as you mentioned - which will be *much* faster.